### PR TITLE
Cleanup makefile

### PR DESCRIPTION
**Pull request by geeknoid**
- Make more human-readable

- Properly track all dependencies

- Simplify container usage

- Combine rules to reduce # tool invocations

- Fix missing python binding for rbac protos.

- Include html link checking as part of linting.

- Retire CircleCi job, since that's now taken care of by the above linting change.

As a result of these changes, a full rebuild takes considerably less time
(1/10th maybe)

### CODE REVIEW

File: python/istio_api/rbac/v1alpha1/rbac_pb2.py [hunk 1]
```diff
@@ -0,0 +1,808 @@
+# -*- coding: utf-8 -*-
```
Comment (howardjohn) [inline python/istio_api/rbac/v1alpha1/rbac_pb2.py]: how come this file, and only this one, is new?

File: Makefile [hunk 2]
```diff
@@ -1,41 +1,33 @@
...
-docker_lock = docker run --rm --user $(uid) -v /etc/passwd:/etc/passwd:ro -v $(pwd):$(repo_mount) -w $(repo_mount) $(lock_img)
-docker_lock_release = docker run --rm --user $(uid) -v /etc/passwd:/etc/passwd:ro -v $(pwd):$(repo_mount) -w $(repo_mount) --entrypoint=/bin/ash $(lock_img) $(repo_mount)/scripts/check-release-locks.sh
-docker_tool = docker run --rm --user $(uid) -v /etc/passwd:/etc/passwd:ro -v $(pwd):$(repo_mount) -w $(repo_mount) $(all_img) prototool
-annotations_prep = docker run --rm --user $(uid) -v /etc/passwd:/etc/passwd:ro -v $(pwd):$(repo_mount) -w $(repo_mount) --entrypoint=/usr/bin/annotations_prep $(gen_img)
+
+protoc = docker run --rm -v $(pwd):$(repo_mount) -w $(mount_dir) $(img) protoc -I/protobuf -I$(repo_dir)
+
+docker = docker run --rm -v $(pwd):$(repo_mount) -w $(repo_mount) $(img)
+protolock = $(docker) protolock
+protolock_release = $(docker) /bin/ash $(repo_mount)/scripts/check-release-locks.sh
```
Comment (sdake) [inline Makefile]: why `bin/ash`?

File: Makefile [hunk 3]
```diff
@@ -54,65 +46,57 @@ importmaps := \
...
-$(config_mcp_pb_pythons): $(config_mcp_protos)
-	## Generate python/istio_api/mcp/v1alpha1/*_pb2.py
-	@$(docker_lock) status
-	@$(docker_gen) $(protoc_gen_python_plugin) $^
+generate-mcp: $(mcp_pb_gos) $(mcp_pb_doc) $(mcp_pb_pythons)
 
 clean-mcp:
-	rm -f $(config_mcp_pb_gos)
-	rm -f $(config_mcp_pb_doc)
+	@rm -fr $(mcp_pb_gos) $(mcp_pb_doc) $(mcp_pb_pythons)
```
Comment (sdake) [inline Makefile]: imo displaying the cmd line in make is a better practice.

File: Makefile [hunk 4]
```diff
@@ -1,41 +1,33 @@
...
-gen_img := gcr.io/istio-testing/protoc:2019-07-26
-lock_img := gcr.io/istio-testing/protolock:2019-03-11
-all_img := gcr.io/istio-testing/api-build-tools:2019-07-26
+img := gcr.io/istio-testing/api-build-tools:2019-07-26
 pwd := $(shell pwd)
 mount_dir := /src
 repo_dir := istio.io/api
 repo_mount := $(mount_dir)/istio.io/api
-uid := $(shell id -u)
-docker_gen := docker run --rm --user $(uid) -v /etc/passwd:/etc/passwd:ro -v $(pwd):$(repo_mount) -w $(mount_dir) $(gen_img) -I$(repo_dir)
```
Comment (sdake) [inline Makefile]: need --user to output as the local user rather than root.  need the -v /etc/passwd to map uids propoerly.

File: Makefile [hunk 5]
```diff
@@ -1,41 +1,33 @@
...
 out_path = .
-docker_lock = docker run --rm --user $(uid) -v /etc/passwd:/etc/passwd:ro -v $(pwd):$(repo_mount) -w $(repo_mount) $(lock_img)
-docker_lock_release = docker run --rm --user $(uid) -v /etc/passwd:/etc/passwd:ro -v $(pwd):$(repo_mount) -w $(repo_mount) --entrypoint=/bin/ash $(lock_img) $(repo_mount)/scripts/check-release-locks.sh
-docker_tool = docker run --rm --user $(uid) -v /etc/passwd:/etc/passwd:ro -v $(pwd):$(repo_mount) -w $(repo_mount) $(all_img) prototool
-annotations_prep = docker run --rm --user $(uid) -v /etc/passwd:/etc/passwd:ro -v $(pwd):$(repo_mount) -w $(repo_mount) --entrypoint=/usr/bin/annotations_prep $(gen_img)
+
+protoc = docker run --rm -v $(pwd):$(repo_mount) -w $(mount_dir) $(img) protoc -I/protobuf -I$(repo_dir)
+
+docker = docker run --rm -v $(pwd):$(repo_mount) -w $(repo_mount) $(img)
```
Comment (sdake) [inline Makefile]: might rename this, as docker is a common word and might confuse people.

File: python/istio_api/rbac/v1alpha1/rbac_pb2.py (hunk 1 above)
Comment (geeknoid) [reply python/istio_api/rbac/v1alpha1/rbac_pb2.py]: Huh, I'm puzzled as to where this came from. I thought the only file in this PR was the makefile. I'll figure it out.

File: Makefile [hunk 6]
```diff
@@ -54,65 +47,57 @@ importmaps := \
...
-	generate-envoy-python \
+	generate-mcp \
+	generate-mesh \
+	generate-mixer \
+	generate-networking \
+	generate-rbac \
+	generate-authn \
+	generate-envoy \
+	generate-policy \
 	generate-annotations \
```
Comment (sdake) [inline Makefile]: need to remove trialing '\'

File: Makefile [hunk 7]
```diff
@@ -150,109 +145,62 @@ mixer_v1_path := mixer/v1
...
-	@$(docker_gen) $(protoc_gen_python_plugin) $^
+generate-mixer: \
+	$(mixer_v1_pb_gos) $(mixer_v1_pb_doc) $(mixer_v1_pb_pythons) \
+	$(mixer_config_client_pb_gos) $(mixer_config_client_pb_doc) $(mixer_config_client_pb_pythons) \
+	$(mixer_adapter_model_v1beta1_pb_gos) $(mixer_adapter_model_v1beta1_pb_doc) $(mixer_adapter_model_v1beta1_pb_pythons)
 
 clean-mixer:
-	rm -f $(mixer_v1_pb_gos) $(mixer_config_client_pb_gos) $(mixer_adapter_model_v1beta1_pb_gos) $(policy_v1beta1_pb_gos) policy/v1beta1/fixed_cfg.pb.go
-	rm -f $(mixer_v1_pb_doc) $(mixer_config_client_pb_doc) $(mixer_adapter_model_v1beta1_pb_doc) $(policy_v1beta1_pb_doc)
+	@rm -fr $(mixer_v1_pb_gos) $(mixer_v1_pb_doc) $(mixer_v1_pb_pythons)
```
Comment (sdake) [inline Makefile]: definately remove the `@` from all commands - makes debugging the gate easier.

File: Makefile (hunk 4 above)
Comment (geeknoid) [reply Makefile]: Readded

File: python/istio_api/rbac/v1alpha1/rbac_pb2.py [hunk 8]
```diff
@@ -0,0 +1,808 @@
...
+# Generated by the protocol buffer compiler.  DO NOT EDIT!
+# source: rbac/v1alpha1/rbac.proto
+
+import sys
```
Comment (sdake) [inline python/istio_api/rbac/v1alpha1/rbac_pb2.py]: no idea if this is correct or not - so not reviewing htis part ;)

File: Makefile (hunk 5 above)
Comment (geeknoid) [reply Makefile]: Renamed to "run".

File: Makefile (hunk 3 above)
Comment (geeknoid) [reply Makefile]: I very  much dislike the pattern. I'd rather have  a quiet command for normal state, and then output for failure. If you want to see more, there's a command-line option for that...

**Comment by sdake** [comment]
I assume the container PR was somewhere modified?  I'd like to take a look at that PR.

File: Makefile (hunk 3 above)
Comment (sdake) [reply Makefile]: cool - what cmd line option - didn't know that.

File: Makefile (hunk 2 above)
Comment (geeknoid) [reply Makefile]: Beats me. I'm just invoking things the way it was...

File: Makefile (hunk 7 above)
Comment (sdake) [reply Makefile]: if there is a cli optino to make to output `@` commands, thats cool, we should be consistent across repos on the rm stage.  I'd like to keep the actual commands free of '@' related to build though.

File: python/istio_api/rbac/v1alpha1/rbac_pb2.py (hunk 1 above)
Comment (geeknoid) [reply python/istio_api/rbac/v1alpha1/rbac_pb2.py]: Turns out I fixed a bug. The previous rbac rule didn't actually build the python files even though it was trying to. Now it does.

File: Makefile (hunk 3 above)
Comment (geeknoid) [reply Makefile]: -n or -d

File: python/istio_api/rbac/v1alpha1/rbac_pb2.py (hunk 8 above)
Comment (geeknoid) [reply python/istio_api/rbac/v1alpha1/rbac_pb2.py]: New file, was incorrectly skipped in previous version of the makefile.

File: Makefile (hunk 2 above)
Comment (geeknoid) [reply Makefile]: I switched to bash, makes things work better.

**Comment by geeknoid** [comment]
PTAL.

While I was in here, I incorporated the linting that was done by CircleCi into the "make lint" rule, which makes it happen via prow. So that's one more repo off of CircleCi.

**Pull request merged**